File: spec/DigitsGuessSpec.js
describe("DigitsGuess", function() {
	
	beforeEach(function(){
		spyOn(DigitsUtil, 'random').andCallThrough();
		DigitsGuess.guess(DigitsUtil.random(), 1234);
	});

	it("random function should have been called after guess is called", function(){
		expect(DigitsUtil.random).toHaveBeenCalled();
	});

	

});

File: spec/digits_guess_spec.rb
require 'digits_guess'
require 'digits_random'

describe DigitsGuess do
  before(:each) do

  end

  it "should return nAnB when guess a number" do
    DigitsRandom.stub(:random).and_return(1234)
    DigitsGuess.guess(DigitsRandom.random, 2314).should == "1A3B"
  end

end